### PR TITLE
Opg

### PR DESCRIPTION
**Pull request by huard**
Tests and docstrings for opg indices

### CODE REVIEW

File: xclim/indices.py [hunk 1]
```diff
@@ -424,9 +531,13 @@ def heat_wave_index(tasmax, thresh=25.0, window=5, freq='YS'):
     Parameters
     ----------
     tasmax : xarrray.DataArray
-      Maximum daily temperature [℃] or [K[
+      Maximum daily temperature [℃] or [K]
     thresh : float
+<<<<<<< HEAD
```
Comment (Zeitsperre) [inline xclim/indices.py]: Merge conflict

File: xclim/utils.py [hunk 2]
```diff
@@ -124,6 +128,192 @@ def daily_downsampler(da, freq='YS'):
     return buffer.groupby('tags')
 
 
+def get_ev_length(ev, verbose=True, method=2, timing=True, using_dask_loop=True):
```
Comment (Zeitsperre) [inline xclim/utils.py]: This is throwing a PEP8 error (C901) due to cyclomatic complexity. Too many functions and conditional statements. We can either suspend that check (as we will likely be writing even more complex indices) or we can try refactoring this function to externalize functions that might be called by other indices in the future. Thoughts?

Comment (huard) [reply xclim/utils.py]: This is @sbiner 's stuff ; ) I'll let him handle it.

File: xclim/indices.py (hunk 1 above)
Comment (huard) [reply xclim/indices.py]: Resolved in the next opg PR...

File: xclim/indices.py [hunk 3]
```diff
@@ -8,6 +8,10 @@
 
 import numpy as np
 import xarray as xr
+from warnings import warn
+import logging
+from xclim.utils import get_ev_length
+from xclim.utils import get_ev_end
```
[the remark this answers is not in the document]
Comment (huard) [reply xclim/indices.py]: Resolved in the next OPG PR.

File: xclim/utils.py (hunk 2 above)
Comment (huard) [reply xclim/utils.py]: I'm resolving since does not affect the current PR. I think I based my PR out of his branch to avoid conflicts.

**Comment by huard** [comment]
@Zeitsperre I'll let you merge since you review is holding it. 
@tlogan2000 we'll need an executive decision on the max consecutive spell question, both on the concept and the code (get_ev_length vs run_length).

**Comment by tlogan2000** [comment]
Short term executive decision is to reprodruce ICCLIM - so resample first then calculate consec days.  Just need to document this is the docstinrg perhaps. 

 I think most use cases of this will be with freq = YS or QS so I think it is not that common a problem....
Longterm we can have many heated discussions ...

**Pull request merged**